posthog/api/survey.py

```diff
@@ -221,19 +221,29 @@ def update(self, instance: Survey, validated_data):
                 existing_flag_serializer.is_valid(raise_exception=True)
                 existing_flag_serializer.save()
             else:
-                new_flag = self._create_new_targeting_flag(instance.name, new_filters)
+                new_flag = self._create_new_targeting_flag(instance.name, new_filters, instance.start_date)
                 validated_data["targeting_flag_id"] = new_flag.id
             validated_data.pop("targeting_flag_filters")
 
+        end_date = validated_data.get("end_date")
+        if instance.targeting_flag:
+            # turn off feature flag if survey is ended
+            if end_date is None:
+                instance.targeting_flag.active = True
+            elif end_date:
```

Comment:
We can be more defensive here and disable the flag whenever an enddate is not None

```suggestion
            else:
```